src/main/java/dev/blaauwendraad/masker/json/KeyMatcher.java

```diff
@@ -207,10 +233,17 @@ private TrieNode searchForJsonPathKeyNode(byte[] bytes, Iterator<MaskingState.Se
             if (node == null) {
                 return null;
             }
-            MaskingState.SegmentReference segmentReference = jsonPath.next();
-            int keyStartIndex = segmentReference.start;
-            int keyLength = segmentReference.offset;
-            for (int i = keyStartIndex; i < keyStartIndex + keyLength; i++) {
+            JsonPathSegmentReference jsonPathSegmentReference = jsonPath.next();
+            int keyOffset = jsonPathSegmentReference.getOffset();
+            int keyStart = jsonPathSegmentReference.getLength();
```

Comment:
Isn't it `keyLength`?

Reply:
It is. I forgot to rename variable name after I renamed the fields.